pkg/products/amqonline/reconciler.go

```diff
@@ -412,6 +419,40 @@ func (r *Reconciler) reconcileAddressSpacePlans(ctx context.Context, serverClien
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
 
+func (r *Reconciler) reconcileServiceAdmin(ctx context.Context, serverClient k8sclient.Client, serviceAdminRole *rbacv1.Role) (integreatlyv1alpha1.StatusPhase, error) {
+	r.logger.Info("reconciling service admin role to the dedicated admins group")
+
+	err := serverClient.Create(ctx, serviceAdminRole)
+	if err != nil && !k8serr.IsAlreadyExists(err) {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("could not create service admin role %v: %w", serviceAdminRole, err)
+	}
+
+	// Bind the amq online service admin role to the dedicated-admins group
+	serviceAdminRoleBinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "dedicated-admins-service-admin",
+			Namespace: r.Config.GetNamespace(),
+		},
+		RoleRef: rbacv1.RoleRef{
+			Name: serviceAdminRole.GetName(),
+			Kind: "Role",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Name: "dedicated-admins",
+				Kind: "Group",
+			},
+		},
+	}
+
+	err = serverClient.Create(ctx, serviceAdminRoleBinding)
```

Comment:
Same here, `controllerutil.CreateOrUpdate` should be preferred.

Reply:
Resolved now, thanks @grdryn